src/parameter/ParameterWidget.cc

```diff
@@ -101,19 +105,37 @@ void ParameterWidget::onSetSaveButton()
 	updateParameterSet(comboBoxPreset->itemData(this->comboBoxPreset->currentIndex()).toString().toStdString());
 }
 
+void ParameterWidget::setFile(QString scadFile){
+	boost::filesystem::path p = scadFile.toStdString();
+	this->jsonFile = p.replace_extension(".json").string();
+}
+
 void ParameterWidget::readFile(QString scadFile)
 {
-	this->jsonFile = scadFile.replace(".scad", ".json").toStdString();
-	bool readable=readParameterSet(this->jsonFile);
-	if(readable){
+	setFile(scadFile);
+	bool exists = boost::filesystem::exists(this->jsonFile);
+	bool writeable = false;
+
+	if(exists){
+		bool readable = readParameterSet(this->jsonFile);
+
+		//check whether file is writeable
+		std::fstream file;
+		file.open(this->jsonFile, std::ios::app);
+		if (file.is_open()) {
+			file.close();
+			writeable = true;
```

Comment:
I think these 5 lines of code could be simplified to smth. like this:
```
if (std::stream{this->jsonFile, std::ios::app}) writeable = true;
```

Reply:
if (std::fstream(this->jsonFile, std::ios::app)) writeable = true;

works, as in:
* it detects, when a json file is read only
* if the json file is not readonly, I can add a new parameter set and save
but I wonder how.

this is using the [constructor of fstream](http://www.cplusplus.com/reference/fstream/fstream/open/). It creates the fstream on the stack. Ah: I found the hint I needed: [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization), the enclosing scope beeing the brackets of the "if".

Neat :-)

Reply:
I was a bit terse in my comment - glad you got it :)
Don't hesitate to ask me to clarify. Hard to gauge what detail level to add..

Reply:
I have the hints I need, thank you. I am always a bit confused where C++ and QT are on the continuum between C and Java (for both of which I have college education). Doesn´t help that a lot of modern C++/QT text books skim over memory management and Stackoverflow tends to use C solutions for C++ problems.

Reply:
These are good reads which cover modern C++:
* A tour of C++ (https://www.amazon.com/Tour-C-Depth/dp/0321958314)
* Effective Modern C++ (e.g. https://edisciplinas.usp.br/pluginfile.php/1995323/mod_resource/content/1/Effective%20Modern%20C%2B%2B%202014.pdf)